openghg/util/_time.py

```diff
@@ -911,3 +913,51 @@ def dates_in_range(
             in_date.append(key)
 
     return in_date
+
+
+def evaluate_sampling_period(sampling_period: Optional[Union[Timedelta, str]]) -> Optional[str]:
+    """
+    Check the sampling period input and convert this into a string containing the
+    sampling period in seconds.
+
+    Args:
+        sampling_period: str or Timedelta value for the time to sample.
+
+    Returns:
+        str : Sampling period as a string containing the number of seconds.
+
+    TODO: Integrate sampling_period handling into logic for time_period (if practical)
+    """
+    # If we have a sampling period passed we want the number of seconds
+    if sampling_period is not None:
+
+        # Check value passed is not just a number with no units
+        try:
```

Comment:
It would probably be better to rewrite this with a regex that detects exactly what we're trying to avoid (integers as strings) e.g. `r"^\s*\d+\s*"` If it is a float (as a string), Timedelta will complain that there is no unit. If it is an int, pandas would treat it as a number of nanoseconds (I think), so that's the case to catch.

Reply:
Thanks for the suggestion - this was mainly a PR to reorganise but I do agree it is worth trying to improve this functionality if/when we have the chance to.

I only spotted your suggested regex after I'd already written one - r"\d+[.]?\d*\s*[a-zA-Z]+" but overall they seem to achieve the same check.